src/core/namespace.c

```diff
@@ -287,10 +288,15 @@ static int apply_mount(
 
                 /* First, get rid of everything that is below if there
                  * is anything... Then, overmount it with an
-                 * inaccessible directory. */
+                 * inaccessible path. */
                 umount_recursive(m->path, 0);
 
-                what = "/run/systemd/inaccessible";
+                r = lstat(m->path, &inaccessible_target);
```

Comment:
Actually I think we should simply fail if the target is a symlink. Following symlinks opens a up a can of worms, and I don't see a particular need to allow that.